#4

maxViolet opened the pull request.

maxViolet committed:
+CollapsingToolbar

maxViolet committed:
+CollapsingToolbar

maxViolet committed:
#3

maxViolet committed:
#3

maxViolet committed:
refactor

maxViolet committed:
исправлены замечания

maxViolet committed:
исправлены замечания

maxViolet committed:
исправлены замечания

maxViolet committed:
исправлены замечания

maxViolet committed:
исправлены замечания

maxViolet committed:
#4

maxViolet committed:
refactor

maxViolet committed:
+added disposable wrapper for Observable
+added OnStart/OnStop/OnDestroy

maxViolet committed:
#2 refactor

user32338 commented on app/src/main/java/com/example/android/maximfialko/NewsListActivity.java:
```diff
@@ -0,0 +1,139 @@
+            return new NewListCallable(NewsItemList.generateNews());
+        })
+                //указываем поток, где будет исполняться код
+                .subscribeOn(Schedulers.io())
```
в задании должна быть задержка + показ прогресса пока грузим

user32338 replied:
у тебя нет

user32338 commented on app/src/main/java/com/example/android/maximfialko/NewsListActivity.java:
```diff
@@ -0,0 +1,139 @@
+                    Log.d("addNews", "Subscribe: " + Thread.currentThread().getName());
+                    getAdapter().add(news.data);
+                }, error -> {
+                    Log.d("onError", error.toString());
```
а пользователь то что должен смотреть в белый экран?

user32338 commented on app/src/main/java/com/example/android/maximfialko/NewsListActivity.java:
```diff
@@ -0,0 +1,139 @@
+    @Override
+    protected void onStop() {
+        super.onStop();
+        disposable = null;
```
э не друг, так нельзя, надо disposable.dispose()

user32338 commented on app/src/main/java/com/example/android/maximfialko/NewsListActivity.java:
```diff
@@ -0,0 +1,139 @@
+    @Override
+    protected void onStart() {
+        super.onStart();
+        loadItems();
```
почему в Start?

user32338 commented on app/src/main/java/com/example/android/maximfialko/NewsListActivity.java:
```diff
@@ -0,0 +1,139 @@
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        adapter = null;
```
это не обязательно) если твоя activity умерет то и адаптер умерт, так как он внутри

user32338 commented on app/src/main/java/com/example/android/maximfialko/data/Margins.java:
```diff
@@ -0,0 +1,48 @@
+     * @param columns number of columns of the RecyclerView
+     */
+    public Margins(@IntRange(from = 0) int margin, @IntRange(from = 0) int columns) {
+        this.margin = margin;
```
что это за отступы, если ты хотел dp то int надо еще преобразовать в него

user32338 reviewed (changes requested):
а где часть 1?

maxViolet committed:
- поправлены замечания:
   1) добавлены задержка в загрузке новостей и ProgressBar
   2) добавлен Util-класс для пересчета (px) в (dp)
   3) правка мелких замечаний
- добавлена папка network для д/з №5 (не относится к данной домашке)

maxViolet committed:
minor bug fixes
 - NPE fix on "disposable = null" in OnsStop()

maxViolet committed:
- refactor
